src/moonlink/src/storage/iceberg/rest_catalog_test.rs

```diff
@@ -105,6 +111,134 @@ async fn test_load_table() {
+        .await
+        .unwrap();
+    let rest_catalog_config = default_rest_catalog_config();
+    let accessor_config = default_accessor_config();
+    let catalog = RestCatalog::new(
+        rest_catalog_config,
+        accessor_config,
+        create_test_table_schema().unwrap(),
+    )
+    .await
+    .unwrap();
+    let ns_parent_ident = guard.namespace.clone().unwrap();
+    assert!(catalog.namespace_exists(&ns_parent_ident).await.unwrap());
+    let ns_name = get_random_string();
+    let ns_ident = NamespaceIdent::from_strs(vec![namespace, ns_name]).unwrap();
+    catalog
+        .create_namespace(&ns_ident, /*properties=*/ HashMap::new())
+        .await
+        .unwrap();
+    assert_eq!(
+        catalog
+            .list_namespaces(Some(&ns_parent_ident))
+            .await
+            .unwrap(),
+        vec![ns_ident.clone()]
+    );
+    catalog.drop_namespace(&ns_ident).await.unwrap();
+    assert_eq!(
+        catalog
+            .list_namespaces(Some(&ns_parent_ident))
+            .await
+            .unwrap(),
+        vec![]
+    );
+    assert!(!catalog.namespace_exists(&ns_ident).await.unwrap());
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_get_namespace() {
+    let namespace = get_random_string();
+    let guard = RestCatalogTestGuard::new(namespace.clone(), /*table=*/ None)
+        .await
+        .unwrap();
+    let rest_catalog_config = default_rest_catalog_config();
+    let accessor_config = default_accessor_config();
+    let catalog = RestCatalog::new(
+        rest_catalog_config,
+        accessor_config,
+        create_test_table_schema().unwrap(),
+    )
+    .await
+    .unwrap();
+    let ns_ident = guard.namespace.clone().unwrap();
+    assert!(catalog.namespace_exists(&ns_ident).await.unwrap());
+    let ns_name = get_random_string();
+    let ns_ident = NamespaceIdent::from_strs(vec![namespace, ns_name]).unwrap();
+    let ns = catalog
+        .create_namespace(&ns_ident, /*properties=*/ HashMap::new())
+        .await
+        .unwrap();
+    assert_eq!(catalog.get_namespace(&ns_ident).await.unwrap(), ns);
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_list_namespace() {
+    fn to_set<T: Eq + Hash>(vec: Vec<T>) -> HashSet<T> {
+        HashSet::from_iter(vec)
+    }
+    let namespace = get_random_string();
+    let guard = RestCatalogTestGuard::new(namespace.clone(), /*table=*/ None)
+        .await
+        .unwrap();
+    let rest_catalog_config = default_rest_catalog_config();
+    let accessor_config = default_accessor_config();
+    let catalog = RestCatalog::new(
+        rest_catalog_config,
+        accessor_config,
+        create_test_table_schema().unwrap(),
+    )
+    .await
+    .unwrap();
+    let namespace_1 = get_random_string();
+    let namespace_2 = get_random_string();
+    let ns_ident_1 = NamespaceIdent::from_strs(vec![namespace.clone(), namespace_1]).unwrap();
```

Comment:
nit: would be nice to check nested namespace as well

eg:
- guard create parent ns
- create ns1, ns2 and list
- create ns inside of ns1, list ns doesn't display